### PR TITLE
Bumping engine-api version (with RegistryAuth fix)

### LINKED ISSUE #2253: Update engine-api version in /vendor
After https://github.com/docker/engine-api/pull/162, lot of function signatures changed in `engine-api`. We should bump up to master soon.

This will have to be done all at once and will touch the following things:
- `engine-api` function calls
- mock and nop clients in Swarm
- unit tests

### PR DESCRIPTION
**Pull request by nishanttotla**
Fixes #2253

Contained in this PR are the following changes
- `engine-api` version bump
- updates to mock and nop clients
- updates `engine-api` function calls that enforce required arguments
- fixes `RegistryAuth` for image pulls

For code review, it should be OK to ignore the first two commits.

### CODE REVIEW

**Comment by allencloud** [comment]
@nishanttotla 
Is https://github.com/docker/swarm/blob/master/cluster/engine.go#L963 still necessary?

**Comment by nishanttotla** [comment]
@allencloud I suspect not (as the comment indicates), but I'm working on it right now (this PR isn't finished yet). Just want to be sure that any existing behavior isn't changed due to the update.

**Comment by allencloud** [comment]
Great Job to use engine-api in Swarm for such a long time of hard working. :)

**Comment by nishanttotla** [comment]
This is ready for review now, ping @docker/core-swarm-maintainers 

Also cc @dhiltgen @alexmavr

**Comment by vieux** [comment]
@nishanttotla should we stay in sync with docker/docker ? 
See https://github.com/docker/docker/blob/master/hack/vendor.sh#L63

The version vendored here is newer than the one used in engine-master

**Comment by nishanttotla** [comment]
@vieux we can do that, although there are only a few commits between here and the vendored version in Docker. The major changes in that gap include the introduction of checkpoints (https://github.com/docker/engine-api/pull/202) and raw inspect functions for networks and volumes (https://github.com/docker/engine-api/pull/228).

If these will be vendored into Docker soon enough, then we can just keep them for now. WDYT?

File: vendor/github.com/docker/engine-api/types/types.go [hunk 1]
```diff
@@ -415,16 +416,16 @@ type VolumeCreateRequest struct {
...
-	ID         string `json:"Id"`
-	Scope      string
-	Driver     string
-	EnableIPv6 bool
-	IPAM       network.IPAM
-	Internal   bool
-	Containers map[string]EndpointResource
-	Options    map[string]string
-	Labels     map[string]string
+	Name       string                      // Name is the requested name of the volume
```
Comment (abronan) [inline vendor/github.com/docker/engine-api/types/types.go]: nit comment: volume? -> network

File: vendor/github.com/docker/engine-api/types/types.go [hunk 2]
```diff
@@ -415,16 +416,16 @@ type VolumeCreateRequest struct {
...
+	ID         string                      `json:"Id"` // ID uniquely indentifies a network on a single machine
+	Scope      string                      // Scope describes the level at which the network exists (e.g. `global` for cluster-wide or `local` for machine level)
+	Driver     string                      // Driver is the Driver name used to create the volume (e.g. `bridge`, `overlay`)
```
Comment (abronan) [inline vendor/github.com/docker/engine-api/types/types.go]: Same here

File: vendor/github.com/docker/engine-api/types/types.go (hunk 1 above)
Comment (nishanttotla) [inline vendor/github.com/docker/engine-api/types/types.go]: @abronan it's actually vendored, so can't fix it here 😅

File: vendor/github.com/docker/engine-api/types/types.go (hunk 2 above)
Comment (nishanttotla) [inline vendor/github.com/docker/engine-api/types/types.go]: Likewise. Looks like someone copy-pasted code 😃

**Comment by vieux** [comment]
LGTM

File: vendor/github.com/docker/engine-api/types/types.go (hunk 1 above)
Comment (abronan) [inline vendor/github.com/docker/engine-api/types/types.go]: Ha indeed, my bad, really went through the whole diff :bowtie:

**Comment by abronan** [comment]
LGTM

**Pull request merged**